manifest.go

```diff
@@ -156,6 +156,29 @@ func validateManifest(s string) ([]error, error) {
 	return warns, nil
 }
 
+// ValidateProjectRoots validates the project roots present in manifest.
+func ValidateProjectRoots(c *Ctx, m *Manifest, sm gps.SourceManager) error {
+	projectRoots := make([]gps.ProjectRoot, 0, len(m.Constraints)+len(m.Ovr))
+	for pr := range m.Constraints {
+		projectRoots = append(projectRoots, pr)
+	}
+	for pr := range m.Ovr {
+		projectRoots = append(projectRoots, pr)
+	}
+
+	for _, pr := range projectRoots {
+		origPR, err := sm.DeduceProjectRoot(string(pr))
+		if err != nil {
+			return errors.Wrapf(err, "could not deduce project root for %s", pr)
+		}
+		if origPR != pr {
+			c.Err.Printf("dep: WARNING: the name for %q in Gopkg.toml should be changed to %q", pr, origPR)
```

Comment:
are we doing the `dep:` prefix anywhere else? i don't recall. i don't think we are.

in any case, though, per your question in the original PR, i think we want to make this a case where we straight up fail, not just warn. there's nothing valid about these, at all, and we should immediately fail out if we encounter them.